### DISCUSSION TITLE
Where would the compute_start_up_script go if you didn't use ghpc to generate the terraform?

### DISCUSSION THREAD

**Discussion by cellis-rxrx** [GoogleCloudPlatform/cluster-toolkit#3058] [Q&A]
Our team using cluster toolkit hasn't been generating the terraform with ghpc, for our own reasons. However we are wondering where the compute startup script would go in this case. 

In this file:

https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/community/modules/scheduler/schedmd-slurm-gcp-v6-controller/variables.tf

I see the following:

variable "compute_startup_script" {
  description = "Startup script used by the compute VMs."
  type        = string
  default     = "# no-op"
}

Would I place the contents of the script in place of "#no op"?

There is also this folder:
https://github.com/GoogleCloudPlatform/cluster-toolkit/tree/main/community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts

However our instance doesn't seem t have that end part "....modules/slurm_files/scripts". Is this because we haven't generated the terraform? Or should we have that there?

And back to the main question, is that where the compute_startup_script would go? In that "scripts" folder?
Thanks for the help!

**Comment by tpdownes** [comment]
Hi @cellis-rxrx 

if you are using the Terraform modules directly, you want to set that input variable to a multiline string containing your actual script. There are a few ways to do that:

1. Use our [modules/scripts/startup-script](https://github.com/GoogleCloudPlatform/cluster-toolkit/tree/main/modules/scripts/startup-script) module which combines runners (shell, ansible, files) into a single startup script uploaded to GCS.
2. Raw text.

You can set the startup_script at the nodeset level, where it only applies to that nodeset or at the `compute_startup_script` on the controller where it applies to all compute nodes. I show the first below:


```hcl
module "compute_nodeset" {
  source                  = "github.com/GoogleCloudPlatform/cluster-toolkit//community/modules/compute/schedmd-slurm-gcp-v6-nodeset?ref=v1.40.0&depth=1"
  allow_automatic_updates = false
  bandwidth_tier          = "gvnic_enabled"
  labels                  = var.labels
  name                    = "compute_nodeset"
  node_count_dynamic_max  = 20
  project_id              = var.project_id
  region                  = var.region
  startup_script = <<-EOT
    #!/bin/bash
    /bin/true
  EOT
  subnetwork_self_link    = module.network.subnetwork_self_link
  zone                    = var.zone
}
```

I am assuming some level of familiarity with defining input variables and using them in your own custom modules.
